apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexService.java

```diff
@@ -925,6 +931,49 @@ public EthDepositsWithOffset getUserFilledOrders(String user, long offset, long
         return dexSmartContractService.getUserFilledOrders(user, offset, limit);
     }
 
+    //TODO implement some abstract function to download all data.
+    public List<EthDepositInfo> getAllUserFilledOrders(String user) throws AplException.ExecutiveProcessException {
+        List<EthDepositInfo> ethDepositInfos = new ArrayList<>();
+        int offset = 0;
+        int limit = 100;
+        int maxPages = 10;
+
+        for (int i = 0; i < maxPages; i++) {
+            EthDepositsWithOffset ethDepositsWithOffset = dexSmartContractService.getUserFilledOrders(user, offset, limit);
+            if (CollectionUtils.isEmpty(ethDepositsWithOffset.getDeposits())) {
```

Comment:
May the collection be null here?

Reply:
right now it's impossible, but who knows what we have tomorrow.